solutions/standard/main.tf

```diff
@@ -49,17 +49,68 @@ module "kms" {
+          standard_key             = false
+          rotation_interval_month  = 3
+          dual_auth_delete_enabled = false
+          force_delete             = true
+        }
+      ]
     }
   ]
 }
 
+#######################################################################################################################
+# COS
+#######################################################################################################################
+
+locals {
+  cos_kms_key_crn   = var.existing_cos_bucket_name != null ? null : local.existing_kms_root_key_id != null ? local.existing_kms_root_key_id : module.kms[0].keys[format("%s.%s", var.cos_key_ring_name, var.cos_key_name)].crn
+  cos_instance_guid = var.existing_cos_instance_crn != null ? element(split(":", var.existing_cos_instance_crn), length(split(":", var.existing_cos_instance_crn)) - 3) : module.cos.cos_instance_guid
+  cos_bucket_name   = var.existing_cos_bucket_name != null ? var.existing_cos_bucket_name : module.cos.bucket_name
+}
+
+module "cos" {
+  source                              = "terraform-ibm-modules/cos/ibm"
+  version                             = "8.2.0"
+  create_cos_instance                 = var.create_cos_instance
+  create_cos_bucket                   = var.create_cos_bucket
+  add_bucket_name_suffix              = var.add_bucket_name_suffix
+  resource_group_id                   = module.resource_group.resource_group_id
+  region                              = var.region
+  cross_region_location               = var.cross_region_location
+  cos_instance_name                   = var.cos_instance_name
+  cos_plan                            = var.cos_plan
+  cos_tags                            = var.cos_instance_tags
+  bucket_name                         = var.cos_bucket_name
+  access_tags                         = var.cos_instance_access_tags
+  management_endpoint_type_for_bucket = var.management_endpoint_type_for_bucket
+  existing_kms_instance_guid          = local.kms_instance_guid
+  kms_key_crn                         = local.cos_kms_key_crn
+  sysdig_crn                          = var.existing_monitoring_crn
+  retention_enabled                   = var.retention_enabled
+  activity_tracker_crn                = var.existing_activity_tracker_crn
+  resource_keys                       = var.resource_keys
+  bucket_cbr_rules                    = var.bucket_cbr_rules
+  instance_cbr_rules                  = var.instance_cbr_rules
+}
+
+
 ########################################################################################################################
 # Event Notifications
 ########################################################################################################################
 
 locals {
   # KMS Related
   existing_kms_instance_crn = var.existing_kms_instance_crn != null ? var.existing_kms_instance_crn : null
+  cos_endpoint              = var.cos_endpoint == null ? "https://${module.cos.s3_endpoint_public}" : var.cos_endpoint
```

Comment:
I would use `var.existing_cos_bucket_name` name here in the logic instead of `var.cos_endpoint`